checkov/cloudformation/checks/resource/BaseCloudsplainingIAMCheck.py

```diff
@@ -79,3 +82,52 @@ def scan_resource_conf(self, conf: dict[str, Any]) -> CheckResult:
     @abstractmethod
     def cloudsplaining_analysis(self, policy: PolicyDocument) -> list[str]:
         raise NotImplementedError()
+
+    def cloudsplaining_enrich_inspected_line(self, policy: PolicyDocument) -> None:
+        try:
+            violating_actions = self.cloudsplaining_analysis(policy)
+            if violating_actions:
+                # in case we have violating actions for this policy we start looking for it through the statements
+                for statement in policy.statements:
+                    actions = statement.statement.get('Action')  # get the actions for this statement
+                    if actions:
+                        if isinstance(actions, str):
+                            actions = [actions]
+                        for action in actions:      # go through the actions of this statement and try to match one violation
+                            for violating_action in violating_actions:
+                                if fnmatch.fnmatch(violating_action, action):      # found the violating action in our list of actions
+                                    action_line = self.extract_action_line(statement.statement, action)
+                                    if action_line > 0:
+                                        self.inspected_key_line = action_line
```

Comment:
so we have only one inspected_key_line in a resource?

Reply:
It will be one inspected key line per violation. This field gets copied to the record, which then gets put in the report.